solc/CommandLineParser.cpp

```diff
@@ -478,32 +477,36 @@ bool CommandLineParser::parseOutputSelection()
 		solAssert(false, "");
 	};
 
-	for (auto&& [optionName, outputComponent]: CompilerOutputs::componentMap())
-		m_options.compiler.outputs.*outputComponent = (m_args.count(optionName) > 0);
-
-	if (m_options.input.mode == InputMode::Assembler && m_options.compiler.outputs == CompilerOutputs{})
+	if(m_options.compiler.outputs.has_value())
```

Comment:
This condition is always false because you never initialize `m_options.compiler.outputs`. This function is actually the right place to do it. And it should have a value only in modes that actually use it (i.e. assembler and compiler).